streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableKTableLeftJoin.java

```diff
@@ -159,11 +160,36 @@ public void init(final ProcessorContext<?, ?> context) {
 
         @Override
         public ValueAndTimestamp<VOut> get(final K key) {
-            final ValueAndTimestamp<V1> valueAndTimestamp1 = valueGetter1.get(key);
+            return computeJoin(key, valueGetter1::get, valueGetter2::get);
+        }
+
+        @Override
+        public ValueAndTimestamp<VOut> get(final K key, final long asOfTimestamp) {
+            return computeJoin(
+                key,
+                k -> valueGetter1.get(k, asOfTimestamp),
+                k -> valueGetter2.get(k, asOfTimestamp));
```

Comment:
As above -- can we get rid of the indirection by calling `get(k, asOfTimestamp)` and pass in the result into `computeJoin()`?